src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala

```diff
@@ -217,105 +217,95 @@ case class ErgoTransaction(override val inputs: IndexedSeq[Input],
       // we check that we're in utxo mode, as eip27Supported flag available only in this mode
       // if we're in digest mode, skip validation
       // todo: this check could be removed after EIP-27 activation
-      if (stateContext.ergoSettings.nodeSettings.stateType.holdsUtxoSet) {
-        lazy val reemissionSettings = stateContext.ergoSettings.chainSettings.reemission
-        lazy val reemissionRules = reemissionSettings.reemissionRules
+      lazy val reemissionSettings = stateContext.ergoSettings.chainSettings.reemission
+      lazy val reemissionRules = reemissionSettings.reemissionRules
 
-        lazy val reemissionTokenId = ModifierId @@ reemissionSettings.reemissionTokenId
-        lazy val reemissionTokenIdBytes = reemissionSettings.reemissionTokenIdBytes
+      lazy val reemissionTokenId = ModifierId @@ reemissionSettings.reemissionTokenId
+      lazy val reemissionTokenIdBytes = reemissionSettings.reemissionTokenIdBytes
 
-        lazy val emissionNftId = ModifierId @@ reemissionSettings.emissionNftId
-        lazy val emissionNftIdBytes = reemissionSettings.emissionNftIdBytes
+      lazy val emissionNftId = ModifierId @@ reemissionSettings.emissionNftId
+      lazy val emissionNftIdBytes = reemissionSettings.emissionNftIdBytes
 
-        lazy val chainSettings = stateContext.ergoSettings.chainSettings
-        lazy val emissionRules = chainSettings.emissionRules
+      lazy val chainSettings = stateContext.ergoSettings.chainSettings
+      lazy val emissionRules = chainSettings.emissionRules
 
-        lazy val height = stateContext.currentHeight
-        lazy val eip27Supported = stateContext.eip27Supported
+      lazy val height = stateContext.currentHeight
 
-        // considering voting for eip27 done, via eip27Supported flag
-        val activationHeight = if (eip27Supported) {
-          reemissionSettings.activationHeight
-        } else {
-          Int.MaxValue
-        }
+      val activationHeight = reemissionSettings.activationHeight
+
+      if (stateContext.currentHeight >= activationHeight) {
+        // reemission check logic below
+        var reemissionSpending = false
+        boxesToSpend.foreach { box =>
+          // checking EIP-27 rules for emission box
+          // for efficiency, skip boxes with less than 100K ERG
+          if (box.value > 100000 * EmissionRules.CoinsInOneErgo) {
+            // on activation height, emissionNft is not in emission box yet, but in injection box
+            if (box.tokens.contains(emissionNftId) ||
+              (height == activationHeight && boxesToSpend(1).tokens.contains(emissionNftId))) {
```

Comment:
`boxesToSpend(1).tokens.contains(emissionNftId)` would be probably better to have as `def` with self-explanatory name... why do we use tokens from the second box?

Reply:
Improved comment on injection box index. Not sure about good name for one-liner